enhancements/alerting/alerting-as-a-feature.md

```diff
@@ -0,0 +1,302 @@
+  - "@wking"
+  - "@cblecker"
+  - "@jharrington22"
+  - "@mwoodson"
+approvers:
+  - TBD
+creation-date: 2021-03-08
+last-updated: 2021-03-08
+status: implementable
+---
+
+# Alerting as a Feature
+
+## Release Signoff Checklist
+
+- [x] Enhancement is `implementable`
+- [x] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [x] Operational readiness criteria is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+## Definitions
+
+Caused Based Alert: Describes the direct cause of an issue. Example: etcdMemoryUtilization @ 100%
+
+Symptom Based Alert = Describes a symptom who's source is a cause based alert. Example: etcd connection latency > 200ms. This may be caused by etcdMemoryUtilization (or not?)
+
+## Summary
+
+Alerting as a feature is a holistic composition of [alerting standards](https://github.com/openshift/enhancements/pull/637) coupled with built-in alerting methodologies (none, symptom based and caused based) with an option to deliver SLOs for a single and/or fleet of clusters. 
+
+## Motivation
+
+OCP as a product does not have built-in [alerting standards](https://github.com/openshift/enhancements/pull/637), nor built-in SLOs driven by symptom or caused based alerting methodologies. This may be acceptable when managing a small number of clusters but at scale (+100) it breaks down. As an SRE of a cluster, you are given alerts and documentation on how to [manage them](https://access.redhat.com/documentation/en-us/openshift_container_platform/4.6/html/monitoring/managing-alerts). This should come "out of the box" to support a natively support fleets.
+
+There are two current efforts in the space at the moment:
+
+* [alerting standards](https://github.com/openshift/enhancements/pull/637) -> this will have component teams own their own components alerts as they are they subject matter experts (caused based alerts). 
+* SLO driven [symptom based alerting](https://docs.google.com/document/d/199PqyG3UsyXlwieHaqbGiWVa8eMWi8zzAn0YfcApr8Q/edit#heading=h.1upja8jlnnwp) is being explored by OSD/SRE. -> this will enable meaningful alerting from a customer perspective in an SLA environment.  
+
+This raises two problems that belong to different domains.
+
+1. Alerting standards provide teams the guidelines to instrument alerts for their components. But this does not guarantee improvement for an SRE.
+2. Symptom Based alerts provide SRE's with user perspective alerting at scale(worth getting out of bed for). But this perspective is not easily illustrated to component engineering teams. 
+
+So how can can these efforts work together in a cohesive life-cycle? 
+
+### Goals
+
+#### Provide two life cycles that deliver Alerting as a feature. 
+
+* Alerting standards are:
+  * defined by engineering, monitoring team and SREs (SRE assist on critical severity only). 
```

Comment:
Alerts are aimed at cluster admins, which means SREs + customers.  There should absolutely be SRE and customer feedback on the utility and actionability of all alerts, not just `critical` alerts.  But the `critical` alerts are the loudest, so that's the most impactful place for folks interested in improving alert quality in general but who only have limited time to invest.

Reply:
+1
This can probably be reworded to incorporate sres generally giving guidance on the alert severity

Reply:
+1. I was trying to illustrate that critical severity involvement should be heavily encouraged, but never as a blocker.